sql-cli/src/sql_cli/sql_directory_parser.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Iterable
+
+import frontmatter
+from sql_cli.utils import find_template_variables
+
+
+class SqlFile:
+    """
+    A SqlFile is equivalent to a transform step in the Astro SDK.
+
+    :param root_directory: The root directory path of the project.
+    :param path: The path to the sql file.
+    :param target_directory: The target directory path for the executable sql.
+    """
+
+    def __init__(
+        self, root_directory: Path, path: Path, target_directory: Path
+    ) -> None:
+        self.root_directory = root_directory
+        self.path = path
+        self.target_directory = target_directory
+
+        post = frontmatter.load(self.path)
+        self.content = post.content
+        self.metadata = post.metadata
+
+    def __eq__(self, __o: object) -> bool:
+        """
+        Check if this sql file equals the given object.
+
+        :param __o: The object to compare to.
+
+        :returns: True if this sql file is equal to the other one.
+        """
+        if isinstance(__o, SqlFile):
+            return self.root_directory == __o.root_directory and self.path == __o.path
+        return False
+
+    def __gt__(self, other: SqlFile) -> bool:
+        """
+        Check if this sql file is greater than the other sql file.
```

Comment:
curious on using the "name" for ">" -- what's the rationale behind it? Depending on that, we might want to use file-size

Reply:
I will document it better even-though I pass the comparison to downstream _pathlib.Path_ comparison. It is about sorting based on name - not size.